setup.ps1

```diff
@@ -0,0 +1,339 @@
+  param(
+    [string]$PythonExecutable
+  )
+
+  $pythonCmd = Get-Command $PythonExecutable -ErrorAction SilentlyContinue
+  if ($pythonCmd) {
+    $command = "$($pythonCmd.Source) --version 2>&1"
+    $versionOutput = Invoke-Expression $command
+    if ($LASTEXITCODE -eq 0) {
+      $version = $versionOutput | Select-String -Pattern "Python (\d+\.\d+\.\d+)" | ForEach-Object { $_.Matches.Groups[1].Value }
+      Write-Log "Python version $version found using executable '$PythonExecutable'."
+      return $true
+    }
+    else {
+      Write-Log "Python executable '$PythonExecutable' not working correctly." -Level 'ERROR'
+      return $false
+    }
+  }
+  else {
+    Write-Log "Python executable '$PythonExecutable' not found." -Level 'ERROR'
+    return $false
+  }
+}
+
+function Find-PythonExecutable {
+
+  $pythonExecutables = @("python", "python3", "python3.9", "python3.10", "python3.11", "C:\Python39\python.exe", "C:\Python310\python.exe", "C:\Python311\python.exe")
+  
+  foreach ($executable in $pythonExecutables) {
+    if (Test-PythonExecutable -PythonExecutable $executable) {
+      return $executable
+    }
+  }
+
+  return $null
+}
+
+# Function to get the list of requirements from a file
+function Get-Requirements($file) {
+  $requirements = @()
+  $lines = Get-Content $file
+  foreach ($line in $lines) {
+    if ($line.StartsWith("-r ")) {
+      $nestedFile = $line.Substring(3).Trim()
+      $nestedFilePath = Join-Path (Split-Path $file -Parent) $nestedFile
+      if (Test-Path $nestedFilePath) {
+        $requirements += Get-Requirements $nestedFilePath
+      }
+    }
+    elseif (-not $line.StartsWith("#")) {
+      $requirements += $line
+    }
+  }
+  return $requirements
+}
+
+function Main {
+  # Exit on lower versions than Python 3.9 or when Python executable not found
+  $pythonExecutable = Find-PythonExecutable
+  if ($null -eq $pythonExecutable) {
+    Write-Host "Error: No suitable Python executable found. Please ensure that Python 3.9 or higher is installed and available in the system PATH."
+    Exit 1
+  }
+
+  "Starting the < operations..." | Out-File $LogFilePath -Append
+
+  # Navigate to the project directory
+  Set-Location -Path $PSScriptRoot
+  "Current directory: $(Get-Location)" | Out-File $LogFilePath -Append
+
+  # Define the path to the Python executable in the virtual environment
+  $VenvPython = "$VenvDir\Scripts\python.exe"
+
+  # Check if the virtual environment exists, if not, create it
+  if (-Not (Test-Path $VenvPython)) {
+    Write-Log "Virtual environment not found. Creating virtual environment..." -Level 'ERROR'
+    & $pythonExecutable -m venv "$VenvName"
+    if (-Not (Test-Path $VenvPython)) {
+      Write-Log "Failed to create virtual environment." -Level 'ERROR'
+      Exit-Script -exitCode 1
+    }
+    Write-Log "Virtual environment created successfully."
+  }
+
+  # Activate the virtual environment
+  $Global:OldVirtualPath = $env:PATH
+  $env:PATH = "$VenvDir\Scripts;$env:PATH"
+
```

Comment:
let me reiterate what i said in https://github.com/freqtrade/freqtrade/pull/10235#discussion_r1610529912

this is NOT the way to activate a virtual environment.

Reply:
I added something to the comment before, but anyways I will stick to the official way now.